lib/vyasa_web/components/contexts/ui_state.ex

```diff
@@ -4,6 +4,48 @@
 # VyasaWeb.Context.Components module.                                           #
 #################################################################################
 
+defmodule VyasaWeb.Context.Components.UiState.Sheaf do
```

Comment:
the ui state for sheaf is a container that encapsulates the ui state of a collection of marks